tests/ops/qubit/test_special_unitary.py

```diff
@@ -93,6 +96,346 @@ def test_pauli_basis_strings(self, n):
+
+    def test_raises_unknown_interface(self):
+        """Test that computing generators raises an error when attempting
+        to use an unknown interface."""
+        with pytest.raises(ValueError, match="The interface test is not supported"):
+            self.get_one_parameter_generators(np.ones(3), 1, "test")
+
+    def test_raises_broadcasting(self):
+        """Test that computing generators raises an error when attempting
+        to use broadcasting."""
+        theta = np.random.random((2, 3))
+        with pytest.raises(ValueError, match="Broadcasting is not supported"):
+            self.get_one_parameter_generators(theta, 1, "dummy")
+
+
+@pytest.mark.parametrize("n", [1, 2])
+class TestGetOneParameterGeneratorsDiffability:
+    """Tests for the effective generators computing function
+    get_one_parameter_generators of qml.SpecialUnitary to be differentiable."""
+
+    @staticmethod
+    def get_one_parameter_generators(theta, num_wires, interface):
+        """Create a SpecialUnitary operation and return its one-parameter group generators."""
+        return qml.SpecialUnitary(theta, list(range(num_wires))).get_one_parameter_generators(
+            interface
+        )
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("use_jit", [True, False])
+    def test_jacobian_jax(self, n, use_jit):
+        """Test that generators are differentiable in JAX."""
+        import jax
+
+        jax.config.update("jax_enable_x64", True)
+        from jax import numpy as jnp
+
+        np.random.seed(14521)
+        d = 4**n - 1
+        theta = jnp.array(np.random.random(d), dtype=jnp.complex128)
+        fn = (
+            jax.jit(self.get_one_parameter_generators, static_argnums=[1, 2])
+            if use_jit
+            else self.get_one_parameter_generators
+        )
+        dOmegas = jax.jacobian(fn, holomorphic=True)(theta, n, "jax")
+        assert dOmegas.shape == (d, 2**n, 2**n, d)
+
+    @pytest.mark.tf
+    def test_jacobian_tf(self, n):
+        """Test that generators are differentiable in Tensorflow."""
+        import tensorflow as tf
+
+        np.random.seed(14521)
+        d = 4**n - 1
+        theta = tf.Variable(np.random.random(d))
+        with tf.GradientTape() as t:
+            Omegas = self.get_one_parameter_generators(theta, n, "tf")
+        dOmegas = t.jacobian(Omegas, theta)
+        assert dOmegas.shape == (d, 2**n, 2**n, d)
+
+    @pytest.mark.torch
+    def test_jacobian_torch(self, n):
+        """Test that generators are differentiable in Torch."""
+        import torch
+
+        np.random.seed(14521)
+        d = 4**n - 1
+        theta = torch.tensor(np.random.random(d), requires_grad=True)
+
+        def fn(theta):
+            return self.get_one_parameter_generators(theta, n, "torch")
+
+        dOmegas = torch.autograd.functional.jacobian(fn, theta)
+        assert dOmegas.shape == (d, 2**n, 2**n, d)
+
+    # Autograd does not support differentiating expm.
+    @pytest.mark.xfail
```

Comment:
Shouldn't we test that the corresponding error is raised?

Reply:
Good call! :)

Reply:
Done.